packages/frontend-api/src/Model/Resolver/Products/Search/ProductSearchQuery.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Model\Resolver\Products\Search;
+
+use Overblog\GraphQLBundle\Definition\Argument;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrontendApiBundle\Component\Validation\PageSizeValidator;
+use Shopsys\FrontendApiBundle\Model\Product\Connection\ProductConnection;
+use Shopsys\FrontendApiBundle\Model\Product\Filter\ProductFilterFacade;
+use Shopsys\FrontendApiBundle\Model\Resolver\AbstractQuery;
+
+class ProductSearchQuery extends AbstractQuery
+{
+    /**
+     * @var \Shopsys\FrontendApiBundle\Model\Resolver\Products\Search\ProductSearchResultsProviderInterface
+     */
+    protected readonly ProductSearchResultsProviderInterface $productSearchResultsProvider;
+
+    /**
+     * @param \Shopsys\FrontendApiBundle\Model\Resolver\Products\Search\ProductSearchResultsProviderResolver $productSearchResultsProviderFacade
+     * @param \Shopsys\FrontendApiBundle\Model\Product\Filter\ProductFilterFacade $productFilterFacade
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     */
+    public function __construct(
+        protected readonly ProductSearchResultsProviderResolver $productSearchResultsProviderFacade,
+        protected readonly ProductFilterFacade $productFilterFacade,
+        protected readonly Domain $domain,
+    ) {
+        $this->productSearchResultsProvider = $this->productSearchResultsProviderFacade->getProductsSearchResultsProviderByDomainId($this->domain->getId());
```

Comment:
it is not a good idea to have "logic" in the constructor - it is considered a code smell as it might have side effects (here e.g. `$this->domain->getId()` might fail on `NoDomainSelectedException` etc.)

Could you please refactor it? You can get rid of the `$productSearchResultsProvider` property completely and just call 
```
$productSearchResultsProvider = $this->productSearchResultsProviderFacade->getProductsSearchResultsProviderByDomainId($this->domain->getId());
```
directly in `productsSearchQuery` method

Reply:
Changed